src/Nox.Generator/Nox.Generator.cs

```diff
@@ -135,6 +114,41 @@ private void GenerateSource(SourceProductionContext context, ImmutableArray<(str
         _debug.GenerateSourceCode();
     }
 
+    private static Dictionary<NoxGeneratorKind, INoxCodeGenerator[]> CreateGeneratorInstances()
```

Comment:
I propose we just scan the executing assembly for types of  INoxCodeGenerator and create them using reflection.

That way this code doesn't need to be maintained by hand and will reduce merge conflicts.

Reply:
I did ask him to do change this, to avoid reflection and Activator.CreateInstance on Build time. So its a trade-off we need to select what we want more!